plans/models.py

```diff
@@ -259,19 +267,17 @@ def extend_account(self, plan, pricing):
             # Processing standard account extending procedure
             if self.plan == plan:
                 status = True
-                if self.expire is None:
-                    pass
-                elif self.expire > date.today():
+                if self.expire is not None and self.expire > date.today():
                     self.expire += timedelta(days=pricing.period)
                 else:
                     self.expire = date.today() + timedelta(days=pricing.period)
 
             else:
                 # This should not ever happen (as this case should be managed by plan change request)
                 # but just in case we consider a case when user has a different plan
-                if self.expire is None:
+                if not self.plan.is_free and self.expire is None:
```

Comment:
This is clearly a bug. The `is_free` is function and should be called with `()`. We should test this.